docs/serial_driver.md

```diff
@@ -68,3 +70,167 @@ You must also enable the ChibiOS `SERIAL` feature:
+
+#### Pin configuration
+
+`SERIAL_USART_TX_PIN` is the USART `TX` pin, `SERIAL_USART_RX_PIN` is the USART `RX` pin. No external pull-up resistors are needed as the `TX` pin operates in push-pull mode. To use this driver the usart peripherals `TX` and `RX` pins must be configured with the correct Alternate-functions. If you are using an STM32F103 everything is already setup, for other MCUs which are using a more modern GPIO configuration you have to specify these by setting `SERIAL_USART_TX_PAL_MODE` and `SERIAL_USART_RX_PAL_MODE`. Refeer to the corresponding datasheets of your MCU or find those Settings in the table below.
+
+#### Connecting the halves and Pin Swap
+Please note that `TX` of the master halve has to be connected with the `RX` pin of the slave halve and `RX` of the master halve has to be connected with the `TX` pin of the slave halve! Usually this pin swap has to be done outside of the MCU e.g. with cables or on the pcb. Some MCUs like the STM32F303 allow this pin swap directly inside the MCU, this feature can be enabled using `#define SERIAL_USART_PIN_SWAP` in your config.h.
+
+#### Setup
+To use the driver, add this to your rules.mk:
+
+```make
+SERIAL_DRIVER = usart_duplex
+```
+
+Next configure the hardware via your config.h:
+
+```c
+#define SERIAL_USART_TX_PIN B6     // USART TX pin
+#define SERIAL_USART_RX_PIN B7     // USART RX pin
+//#define SERIAL_USART_PIN_SWAP      // Swap TX and RX pins if keyboard is master halve.
+                                   // Check if this feature is necessary with your keyboard design and available on the mcu.
+#define SELECT_SOFT_SERIAL_SPEED 1 // or 0, 2, 3, 4, 5
+                                   //  0: about 460800 baud
+                                   //  1: about 230400 baud (default)
+                                   //  2: about 115200 baud
+                                   //  3: about 57600 baud
+                                   //  4: about 38400 baud
+                                   //  5: about 19200 baud
+#define SERIAL_USART_DRIVER UARTD1 // USART driver of TX and RX pin. default: UARTD1
+#define SERIAL_USART_TX_PAL_MODE 7 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 7
+#define SERIAL_USART_RX_PAL_MODE 7 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 7
+#define SERIAL_USART_TIMEOUT 100 // USART driver timeout. default 100
+```
+
+You must also enable the ChibiOS `UART` with blocking api feature:
+* In your board's halconf.h: `#define HAL_USE_USAR TRUE` and `#define UART_USE_WAIT TRUE`
+* In your board's mcuconf.h: `#define STM32_UART_USE_USARTn TRUE` (where 'n' matches the peripheral number of your selected USART on the MCU)
+
+Do note that the configuration required is for the `UART` peripheral, not the `SERIAL` peripheral.
+
+#### Pins for USART Peripherals with Alternate Functions for selected STM32 MCUs
```

Comment:
Would be nice if all of this was a single table, though I know what may be complex to do.

Reply:
Merged the tables into a single one for each mcu. Having one big table for all mcus is not practical imho.